resources/user.go

```diff
@@ -0,0 +1,242 @@
+// Mgmt
+// Copyright (C) 2013-2017+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+	"os/user"
+
+	"github.com/purpleidea/mgmt/recwatch"
+
+	errwrap "github.com/pkg/errors"
+)
+
+func init() {
+	RegisterResource("user", func() Res { return &UserRes{} })
+}
+
+// UserRes is a user account resource
+type UserRes struct {
+	BaseRes `yaml:",inline"`
+	State   string `yaml:"state"` // state: exists, absent
+	UID     string `yaml:"uid"`
```

Comment:
Uid*
(to match same pattern as Gid)

Reply:
Both of these should be the right kind of uint's (look up in the golang docs for the size, and see what /etc/passwd allows as max int) it's probably uint64, but I didn't check, not string